pkg/api/meta/finalizers.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package meta
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// GetFinalizers gets the list of finalizers on obj
+func GetFinalizers(obj runtime.Object) ([]string, error) {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, err
+	}
+	return accessor.GetFinalizers(), nil
+}
+
+// AddFinalizer adds value to the list of finalizers on obj
+func AddFinalizer(obj runtime.Object, value string) error {
```

Comment:
yep - there definitely is. I'd like to roll with this implementation for now, and do the following if we find that it's unstable or broken on some systems (note that TPR storage will still most likely be labelled an alpha feature after we release our first beta):

1. On a `GuaranteedUpdate` call with the deletion timestamp is set (this is a soft-delete), set an appropriate annotation on the TPR (not a finalizer)
2. On `Delete` call, simply delete the TPR on the core API server (regardless of whether we're doing a soft or hard delete)

Thoughts?